### PR TITLE
Enable token exchange

### PR DESCRIPTION
**Pull request by davidfestal**
## Description

This PR allows enabling the Keycloak Token Exchange feature in the Che setup when the OpenShift OAuth integration is enabled.

This allows grabbing a Che authentication token from the OpenShift token of a user, making it easier to integrate Che with third-party tools running in a overall OpenShift environment.

## Related issues

https://github.com/eclipse/che/issues/14390

## Linked Documentation Pull Request:

https://github.com/eclipse/che-docs/pull/826

### CODE REVIEW

**Comment by davidfestal** [comment]
@themr0c Could you have a look to the attached Docs PR and tell if it contains sufficiently complete and consistent content for the docs to finalize ?

I'll wait for your confirmation before merging this PR.

cc  @rkratky

**Comment by davidfestal** [comment]
cc @supriya-bharadwaj

**Comment by davidfestal** [comment]
cc @pmkovar

File: README.md [hunk 1]
```diff
@@ -100,15 +100,16 @@ You can set it in Run configuration in the IDE, or export this env before execut
 
 This applies both to Run and Debug.
 
-### Pre-Reqs: /tmp/keycloak_provision file
+### Pre-Reqs: /tmp/keycloak_provision and oauth_provision files
 
-The operator grabs this file and replaces values to get a string used as exec command to create Keycloak realm, client and user.
+The operator grabs these files and replaces values to get a string used as exec command to configure Keycloak.
```
Comment (rkratky) [inline README.md]:
```suggestion
The operator takes these files and replaces values to get a string used as the `exec` command to configure Keycloak.
```

File: README.md [hunk 2]
```diff
@@ -100,15 +100,16 @@ You can set it in Run configuration in the IDE, or export this env before execut
...
 Make sure you run the following before running/debugging:
 
 ```
 cp deploy/keycloak_provision /tmp/keycloak_provision
+cp deploy/oauth_provision /tmp/oauth_provision
 ```
-This file is added to a Docker image, thus this step isn't required when deploying an operator image.
+These files are added to a Docker image, thus this step isn't required when deploying an operator image.
```
Comment (rkratky) [inline README.md]:
```suggestion
These files are added to a container image, and thus this step is not required when deploying an operator image.
```

**Comment by davidfestal** [comment]
@rkratky did you have a chance to also give a look to the linked docs PR eclipse/che-docs
https://github.com/eclipse/che-docs/pull/826, and confirm the content is complete / consistent enough ?

**Comment by rkratky** [comment]
Pardon my ignorance, @davidfestal :) Docs PR reviewed.

**Comment by davidfestal** [comment]
Thanks @rkratky !

**Pull request merged**